tutorials/getting_started/another_model_example.md

```diff
@@ -0,0 +1,611 @@
+  "iterator": {
+    "type": "bucket",
+    "sorting_keys": [["abstract", "num_tokens"], ["title", "num_tokens"]],
+    "batch_size": 64
+  },
+  "trainer": {
+    "num_epochs": 40,
+    "patience": 10,
+    "cuda_device": 0,
+    "grad_clipping": 5.0,
+    "validation_metric": "+accuracy",
+    "optimizer": {
+      "type": "adagrad"
+    }
+  }
+```
+
+These parameters specify what data to use for training and validation, how to group the data into
+batches (that's the "iterator" key), and how to train the model.  The data iterator that we're
+using sorts the instances by the number of tokens in the abstract, then the number of tokens in the
+title (with a little bit of noise added), then groups them into batches of size 64.  The sorting
+helps with efficiency, so you don't waste too much computation on padding tokens.  The trainer uses
+the AdaGrad optimizer for 40 epochs, stopping if validation accuracy has not increased for the last
+10 epochs.  The data files themselves are not included in the repository, but you can download the
+data from the [Semantic Scholar website](http://labs.semanticscholar.org/corpus/), do whatever
+heuristic venue labeling you want, and split it into train and dev.
+
+The last piece of the configuration to look at is the model itself:
+
+```json
+  "model": {
+    "type": "paper_classifier",
+    "text_field_embedder": {
+      "tokens": {
+        "type": "embedding",
+        "pretrained_file": "https://s3-us-west-2.amazonaws.com/allennlp/datasets/glove/glove.6B.100d.txt.gz",
+        "embedding_dim": 100,
+        "trainable": false
+      }
+    },
+    "title_encoder": {
+      "type": "lstm",
+      "bidirectional": true,
+      "input_size": 100,
+      "hidden_size": 100,
+      "num_layers": 1,
+      "dropout": 0.2
+    },
+    "abstract_encoder": {
+      "type": "lstm",
+      "bidirectional": true,
+      "input_size": 100,
+      "hidden_size": 100,
+      "num_layers": 1,
+      "dropout": 0.2
+    },
+    "classifier_feedforward": {
+      "input_dim": 400,
+      "num_layers": 2,
+      "hidden_dims": [200, 3],
+      "activations": ["relu", "linear"],
+      "dropout": [0.2, 0.0]
+    }
+   },
+```
+
+Here again the "type" key matches what we registered our `Model` as, and the rest of the parameters
+get passed to our `AcademicPaperClassifier.from_params` method.  We have parameters for
+constructing the `TextFieldEmbedder`, which says to take the out of the "tokens" `TokenIndexer` and
+embed it using fixed, pretrained glove vectors.  If I wanted to also use a CNN over character ids,
+I would need to add a `TokenCharactersIndexer` to the `TokenIndexers` in the `DatasetReader`, and
+a corresponding entry to the `TextFieldEmbedder` parameters specifying a `TokenCharactersEncoder`.
+
+The title and abstract encoders are both bi-directional LSTMs.  The actual LSTM implementation
+here is a thin wrapper around pytorch's built-in LSTMs that make them conform to the
```

Comment:
It's not thin haha

Reply:
Good point =).  Getting fatter by the day...